core/src/main/java/com/linecorp/armeria/common/RequestContext.java

```diff
@@ -276,6 +276,25 @@ default EventLoop contextAwareEventLoop() {
         return new RequestContextAwareEventLoop(this, eventLoop());
     }
 
+    /**
+     * Pushes this context to the thread-local stack if there is no current context. If there is, this method
+     * will throw IllegalStateException. To pop the context from the stack, call {@link SafeCloseable#close()},
+     * which can be done using a {@code try-with-resources} block.
+     */
+    default SafeCloseable propagateContextIfNotPresent() {
```

Comment:
- How about `pushIfAbsent()`?
- I'm also curious if we should add `RequestContext.push()` that deprecates `RequestContext.push(RequestContext)`. What do you think? /cc @anuraaga